akka-docs/src/main/paradox/stream/operators/ActorFlow/ask.md

```diff
@@ -28,16 +28,15 @@ If any of the asks times out it will fail the stream with an @apidoc[AskTimeoutE
 The `ask` operator requires
 
 * the actor `ref`,
-* a `makeMessage` function to create the message sent to the actor from the incoming element and the actor ref accepting the actor's reply message,
-* and a timeout.
+* a `makeMessage` function to create the message sent to the actor from the incoming element, and the actor ref accepting the actor's reply message and a timeout.
 
 See also:
 
 * @ref[Flow.ask](../Source-or-Flow/ask.md) for the classic actors variant
 
 ## Examples
 
-The `ActorFlow.ask` sends a message to the actor which expects `Asking` messages which contain the actor ref for replies of type `Reply`. The replies are emitted when received and the `map` extracts the message `String`.
+The `ActorFlow.ask` sends a message to the actor which expects `Asking` messages which contain the actor ref for replies of type `Reply`. When it receives a reply, it emits it  and the `map` extracts the message `String`.
```

Comment:
I'm not sure it is very clear what "it" is here. Perhaps:

```suggestion
The `ActorFlow.ask` sends a message to the actor which expects `Asking` messages which contain the actor ref for replies of type `Reply`. The replies are emitted by the `askFlow` and the `map` extracts the message `String`.
```

Reply:
Maybe we can even remove the double-`which` and some pronouns:
```suggestion
The `ActorFlow.ask` sends a message to the actor. The actor expects `Asking` messages which contain the actor ref for replies of type `Reply`. When the actor for replies receives a reply, the `ActorFlow.ask` stream stage emits the reply and the `map` extracts the message `String`.
```